src/axom/inlet/Table.hpp

```diff
@@ -19,17 +19,225 @@
 #include <functional>
 #include <unordered_map>
 #include <tuple>
+#include <type_traits>
+
+#include "fmt/fmt.hpp"
 
 #include "axom/inlet/Field.hpp"
 #include "axom/inlet/Reader.hpp"
 #include "axom/inlet/SchemaCreator.hpp"
 
 #include "axom/sidre.hpp"
 
+/*!
+ *******************************************************************************
+ * \brief Prototype for user-defined types wishing to define a by-value read
+ * from inlet with axom::inlet::Table::get(const std::string&)
+ * 
+ * This is the only way of reading in a non-default-constructible type from
+ * Inlet.
+ * 
+ * \see axom::inlet::Table::get(const std::string&)
+ *******************************************************************************
+ */
+template <typename T>
+struct FromInlet
+{
+  // "Poison" the base implementation so it cannot be used
+  // and so it can be checked in a SFINAE context
+  FromInlet() = delete;
+};
+
 namespace axom
 {
 namespace inlet
 {
+namespace detail
+{
+/*!
+ *******************************************************************************
+ * \class is_inlet_primitive
+ *
+ * \brief A type trait for checking if a type is isomorphic to a Lua primitive
```

Comment:
I think `Lua` should be removed from here, and "inlet primitive" should be properly defined (here or elsewhere).